File: .tmp_py4013/batch_11_dump/3224_minimum_array_changes_to_make_differences_equal.js
// LeetCode 3224 - Minimum Array Changes to Make Differences Equal
// https://leetcode.com/problems/minimum-array-changes-to-make-differences-equal/

var minChanges = function(nums, k) {
    const d = new Array(k + 2).fill(0);
    const n = nums.length;
    for (let i = 0; i < n / 2; i++) {
        let x = nums[i], y = nums[n - 1 - i];
        if (x > y) { const t = x; x = y; y = t; }
        d[0] += 1;
        d[y - x] -= 1;
        d[y - x + 1] += 1;
        const mx = Math.max(y, k - x);
        d[mx + 1] -= 1;
        d[mx + 1] += 2;
    }
    let ans = n, s = 0;
    for (const x of d) {
        s += x;
        ans = Math.min(ans, s);
    }
    return ans;
};

File: .tmp_rs4013/rust_check15/3224_minimum_array_changes_to_make_differences_equal.rs
struct Solution;
// LeetCode 3224 - Minimum Array Changes to Make Differences Equal
// https://leetcode.com/problems/minimum-array-changes-to-make-differences-equal/

impl Solution {
    pub fn min_changes(nums: Vec<i32>, k: i32) -> i32 {
        let mut d = vec![0; (k + 2) as usize];
        let n = nums.len();
        for i in 0..n / 2 {
            let mut x = nums[i];
            let mut y = nums[n - 1 - i];
            if x > y {
                std::mem::swap(&mut x, &mut y);
            }
            d[0] += 1;
            d[(y - x) as usize] -= 1;
            d[(y - x + 1) as usize] += 1;
            let mx = y.max(k - x);
            d[(mx + 1) as usize] -= 1;
            d[(mx + 1) as usize] += 2;
        }
        let mut ans = n as i32;
        let mut s = 0;
        for x in d {
            s += x;
            ans = ans.min(s);
        }
        ans
    }
}

fn main() {}
